### PR TITLE
Add: Dropdown Component

### PR DESCRIPTION
**Pull request by yozzyurubuza**
Description:
Implemented a new component that can display any given data from an array of object that follows the format of label:value.

Also applied useEffect and useRef of Hooks System and added comments to give a better understanding of the written code and its purpose.

### CODE REVIEW

File: 12-hooks-in-react/src/components/Dropdown.js [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+import React, { useState, useEffect, useRef } from "react";
 
 const Dropdown = ({ options, selected, onSelectedChange }) => {
   const [open, setOpen] = useState(false);
+  const ref = useRef();
+
+  useEffect(() => {
+    const onBodyClick = (event) => {
+      // If user click in the dropdown, do nothing in this function.
+      if (ref.current.contains(event.target)) return;
```
Comment (jcgalbano) [inline 12-hooks-in-react/src/components/Dropdown.js:10]: Good use of guard clause, remember this: return early

**Review by jcgalbano — CHANGES_REQUESTED** [review]
Good job just a few comments

File: 12-hooks-in-react/src/components/Dropdown.js (hunk 1 above)
Comment (yozzyurubuza) [reply 12-hooks-in-react/src/components/Dropdown.js:10]: Thanks! :100:

File: 12-hooks-in-react/src/components/Dropdown.js [hunk 2]
```diff
@@ -1,7 +1,28 @@
-import React, { useState } from "react";
+import React, { useState, useEffect, useRef } from "react";
 
 const Dropdown = ({ options, selected, onSelectedChange }) => {
   const [open, setOpen] = useState(false);
+  const ref = useRef();
+
+  useEffect(() => {
+    const onBodyClick = (event) => {
```
[the remark this answers is not in the document]
Comment (yozzyurubuza) [reply 12-hooks-in-react/src/components/Dropdown.js]: Done, refactored the function outside.

**Review by jcgalbano — APPROVED** [review]

**Pull request merged**